modulemd/include/private/modulemd-util.h

```diff
@@ -26,33 +26,99 @@ G_BEGIN_DECLS
  */
 
 
+/**
+ * MODULEMD_ERROR:
+ *
+ * A convenience macro for identifying an error in the general modulemd domain.
+ *
+ */
 #define MODULEMD_ERROR modulemd_error_quark ()
+
+/**
+ * modulemd_error_quark:
+ *
+ * Returns: A #GQuark used to identify an error in the general modulemd domain.
+ *
+ * Since: 2.0
+ */
 GQuark
 modulemd_error_quark (void);
 
-enum ModulemdError
+/**
+ * ModulemdErrorEnum:
+ * @MODULEMD_ERROR_UPGRADE: Represents an error encountered while upgrading the
+ * metadata version of a module stream or module defaults.
+ * @MODULEMD_ERROR_VALIDATE: Represents an error encountered while validating
+ * module metadata.
+ * @MODULEMD_ERROR_FILE_ACCESS: Represents an error encountered when attempting
+ * to access a file.
+ * @MODULEMD_ERROR_NO_MATCHES: Represents an error indicating that no streams
+ * matched when searching for a specific module stream. Since: 2.2
+ * @MODULEMD_ERROR_TOO_MANY_MATCHES: Represents an error indicating that
+ * multiple streams matched when searching for a specific module stream.
+ * Since: 2.2
+ *
+ * Since: 2.0
+ */
+typedef enum
 {
   MODULEMD_ERROR_UPGRADE,
   MODULEMD_ERROR_VALIDATE,
   MODULEMD_ERROR_FILE_ACCESS,
   MODULEMD_ERROR_NO_MATCHES,
   MODULEMD_ERROR_TOO_MANY_MATCHES
-};
+} ModulemdErrorEnum;
 
+/**
+ * modulemd_tracer:
+ * @function_name: The name of the function to be tracked by this
+ * #modulemd_tracer.
+ *
+ * Since: 2.0
+ */
 typedef struct _modulemd_tracer
 {
   gchar *function_name;
 } modulemd_tracer;
 
+/**
+ * modulemd_trace_init:
```

Comment:
I'd like to see this comment indicate that it's not recommended to use this function directly. Instead, `MODULEMD_INIT_TRACE` should be preferred.